src/ray/core_worker/reference_count.cc

```diff
@@ -256,7 +256,9 @@ void ReferenceCounter::SetNestedRefInUseRecursive(ReferenceTable::iterator inner
 
 void ReferenceCounter::ReleaseAllLocalReferences() {
   absl::MutexLock lock(&mutex_);
-  for (auto &ref : object_id_refs_) {
+  /// NOTE: A copy here to avoid the risk of iterator invalidation issue.
+  auto all_refs_copy = object_id_refs_;
```

Comment:
This might be pretty expensive to copy all Reference?

What about just

```cpp
std::vector<ObjectID> ref_to_remove;
for (const &auto ref : object_id_refs) {
    for (int i = ref.second.local_ref_count; i > 0; --i) {
      ref_to_remove.push_back(ref.first);
    }
}
for (const auto &object_id_to_remove : ref_to_remove) {
    RemoveLocalReferenceInternal(object_id_to_remove, nullptr);
}
```

Reply:
This method only be invoked when process exiting, so I believe it's not matter.
Both are fine to me actually.